torch/_functorch/_aot_autograd/functional_utils.py

```diff
@@ -166,9 +167,12 @@ def has_metadata_mutation(f_arg, arg, *, check_only_storage_mutation: bool):
         # it experiences an data mutation, we pessimistically think that the set_()
         # call is necessary here. We could in theory fix this, but this will
         # hopefully never happen in user code, and is not needed for fsdp.
-        same_storages = StorageWeakRef(arg.untyped_storage()) == StorageWeakRef(
-            arg_after.untyped_storage()
-        )
+        if is_sparse_any(arg):
+            same_storages = False
```

Comment:
This should likely be a TODO item that involves adding sparse tensors support to functionalization.